Documentation/storage-providers.md

```diff
@@ -0,0 +1,89 @@
+grows, we expect this framework to grow as common storage constructs are identified
+that will benefit the community. Rook does not aim to replace other frameworks or
+communities, but to fill gaps not filled by other core projects. Here are
+some examples of other frameworks areas that already meet many community needs:
+
+* [Operator SDK](https://sdk.operatorframework.io/): For creating general operators.
+* [Storage SIG](https://github.com/kubernetes/community/tree/master/sig-storage):
+  For K8s storage API included in K8s, CSI drivers, snapshotting, local storage, and much more.
+* [Controller Runtime](https://github.com/kubernetes-sigs/controller-runtime):
+  Set of libraries for building Controllers for K8s.
+
+Rook does not aim to be a general framework for storage, but to provide a
+very specific set of helper to meet the storage provider needs in the Rook project.
+
+## Rook Framework
+
+Rook provides the following framework to assist storage providers in building an operator:
+
+* Common golang packages shared by storage providers are in the main [Rook repo](https://github.com/rook/rook).
+* Common build scripts for building the operator images are in the main
+  [Rook repo](https://github.com/rook/rook/tree/master/build)].
+* Each provider has its own repo under the [Rook org](https://github.com/rook).
+  * Multiple community members are given push access to the repo, including
+    owners of the storage provider, Rook steering committee members,
+    and other Rook maintainers if deemed helpful or necessary by the steering
+    committee. Maintainers for the new provider are added according to the
+    [governance](https://github.com/rook/rook/blob/master/GOVERNANCE.md).
+  * Providers added to Rook prior to 2020 are grandfathered into the main
+    [Rook repo](https://github.com/rook/rook).
+* Storage providers must follow the Rook [governance](https://github.com/rook/rook/blob/master/GOVERNANCE.md)
+  in the interest of the good of the overall project. Storage providers have
+  autonomy in their feature work, while collaboration
+  with the community is expected where there are shared features.
+* A quarterly release cadence is in place for the operators in the main Rook repo.
+  Operators in their own repo define their own cadence and versioning scheme as desired.
+  * Storage providers own their release process, while following Rook best practices to
+    ensure high quality.
+* Docker images are pushed to the [Rook DockerHub](https://hub.docker.com/u/rook) where
+  each storage provider has its own repo.
+* Documentation for the storage provider is to be written by the storage provider
+  members. The build process will publish the documentation to the [Rook website](https://rook.github.io/docs/rook/master/).
+* All storage providers are added to the [Rook.io website](https://rook.io/)
+* A great Slack community is available where you can rally your troops.
```

Comment:
A mention about each provider having its own CI and that GitHub actions are recommended due to Rook's graduated status?

Reply:
I mentioned the separate release process, but will also add a comment about the independent CI and GitHub actions. But how does this relate to graduation?

Reply:
@leseb, a comment about CI and GitHub actions has been added. So is this resolved? 

But I'd also be interested: why is graduation status the reason for recommending the use of GitHub actions?

Reply:
Maybe there is some confusion here, the graduated status gave us more
worker/higher rate limit. That's it :)